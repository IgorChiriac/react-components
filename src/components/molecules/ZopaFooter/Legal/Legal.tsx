import React from 'react';
import styled from 'styled-components';

import * as colors from '../../../../constants/colors';
import Separator from '../Separator/Separator';
import Wrapper from '../Wrapper/Wrapper';

const StyledParagraph = styled.p`
  color: ${colors.neutral.neutral400};
  font-size: 11px;
  line-height: 1.5;
  margin: 0;
`;

const Legal = () => {
  const { about, rightsReserved } = {
    about: `Zopa Limited is incorporated in England & Wales (registration number 05197592), with its registered office at 1st Floor, Cottons Centre, Tooley Street, 
    London, SE1 2QG. Zopa Limited is authorised and regulated by the Financial Conduct Authority, 
    and entered on the Financial Services Register under firm registration number 718925.`,

    rightsReserved: `© Zopa Limited ${new Date().getFullYear()} All rights reserved. 'Zopa' and the Zopa logo are trade marks of Zopa Limited. 
    Zopa is a member of CIFAS – the UK's leading anti-fraud association, 
    and we are registered with the Office of the Information Commissioner (No. Z8797078).`,
  };

  return (
    <>
      <Wrapper>
        <StyledParagraph>{about}</StyledParagraph>
      </Wrapper>
      <Separator />
      <Wrapper>
        <StyledParagraph>{rightsReserved}</StyledParagraph>
      </Wrapper>
    </>
  );
};

export default Legal;
