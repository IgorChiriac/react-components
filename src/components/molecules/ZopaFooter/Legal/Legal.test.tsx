import React from 'react';
import { render } from '@testing-library/react';
import Legal from './Legal';

describe('<Legal />', () => {
  it('renders the component', async () => {
    const { container } = render(<Legal />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
