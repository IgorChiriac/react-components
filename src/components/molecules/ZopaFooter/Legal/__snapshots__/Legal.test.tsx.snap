// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Legal /> renders the component 1`] = `
.c0 {
  padding: 18px 0;
}

.c1 {
  color: #6F748B;
  font-size: 11px;
  line-height: 1.5;
  margin: 0 0 24px 0;
}

<div>
  <div
    class="c0"
  >
    <p
      class="c1"
    >
      Zopa Limited is authorised and regulated by the Financial Conduct Authority, and entered on the Financial Services Register (718925). Zopa Bank Limited is authorised by the Prudential Regulation Authority and regulated by the Financial Conduct Authority and the Prudential Regulation Authority, and entered on the Financial Services Register (800542). Zopa Limited (05197592) and Zopa Bank Limited (10627575) are both incorporated in England & Wales and have their registered office at: 1st Floor, Cottons Centre, Tooley Street, London, SE1 2QG.
    </p>
    <p
      class="c1"
    >
      © Zopa Bank Limited 2016 All rights reserved. 'Zopa' is a trademark of Zopa Bank Limited.
    </p>
    <p
      class="c1"
    >
      Zopa is a member of Cifas – the UK’s leading anti-fraud association, and we are registered with the Office of the Information Commissioner (ZA275984, Z8797078).

    
    </p>
  </div>
</div>
`;
